QA.md

```diff
@@ -0,0 +1,210 @@
+# Quality Assurance
+
+This document provides an overview of our quality assurance processes for
+Tendermint. It maps out the various concerns we have regarding software quality
+against the approaches we use to ensure quality. It also attempts to capture the
+current state of our QA process and where there are still deficiencies to be
+addressed.
+
+There is a [special QA log section](#qa-log) at the end of this document where
+we can capture references to one-off QA efforts.
+
+## Approaches
+
+We make use of the following approaches to ensure delivery of quality software,
+with associated relative costs where applicable:
+
+| Approach                  | Setup Cost   | Per-Execution Cost |
+|---------------------------|--------------|--------------------|
+| Code review               |              | Moderate/High      |
+| Static analysis           | Low          | Low                |
+| Unit testing              | Moderate     | Low                |
+| Fuzz testing              | Low/Moderate | Low                |
+| Model-based testing (MBT) | High         | Low                |
+| End-to-end (E2E) testing  | High         | Low                |
+| Testnet testing           | High         | Moderate/High      |
+| Protocol audits           |              | High               |
+
```

Comment:
Consider adding an entry here with something like "Protocol/Interfate Specification" (setup cost high, "per-execution cost" empty). Or, if you don't consider it an "approach", this info should be somewhere in this document.

This is needed for, at least, MBT and protocol audits mentioned above. Although, theoretically, it would also be needed to be able to say is a test (at whatever level) is passing or not.